test/dependency-manager/interfaces.test.ts

```diff
@@ -41,19 +45,29 @@ describe('interfaces', function () {
         "CHECKOUT_ADDR": {
           "valueFrom": {
             "dependency": "architect/checkout",
+            "interface": "main",
             "value": "$HOST:$PORT"
           }
         },
         "CHECKOUT_ADDR_DEFAULT": {
           "default": {
             "valueFrom": {
               "dependency": "architect/checkout",
+              "interface": "main",
               "value": "$HOST:$PORT"
             }
           }
+        },
+        "HOST": {
+          "default": {
+            "valueFrom": {
+              "self": true,
```

Comment:
If you don't specify a dependency and you specify an interface - I think it should be inferred that you're self referencing. 
@davidthor thoughts?

Reply:
I concur. Self references are default